include/dm.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef DM_H_
+#define DM_H_
+
+/**@file
+ * @defgroup dm Distance Measurement API
+ * @{
+ * @brief API for the Distance Measurement (DM).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+#include <stdlib.h>
+#include <zephyr/types.h>
+#include <bluetooth/addr.h>
+
+/** @brief Role definition. */
+enum dm_dev_role {
+	/* Undefined role*/
+	DM_ROLE_NONE,
+
+	/* Act as a initiator */
+	DM_ROLE_INITIATOR,
+
+	/* Act as a reflector */
+	DM_ROLE_REFLECTOR,
+};
+
+/** @brief Ranging mode definition. */
+enum dm_ranging_mode {
+	/* Round trip timing */
+	DM_RANGING_MODE_RTT,
+
+	/* Multi-carrier phase difference */
+	DM_RANGING_MODE_MCPD,
+};
+
+/** @brief Measurement quality definition. */
+enum dm_quality {
+	/* Good measurement quality */
+	DM_QUALITY_OK,
+
+	/* Poor measurement quality */
+	DM_QUALITY_POOR,
+
+	/* Measurement not for use */
+	DM_QUALITY_DO_NOT_USE,
+
+	/* Incorrect CRC measurement */
+	DM_QUALITY_CRC_FAIL,
+
+	/* Measurement quality not specified */
+	DM_QUALITY_NONE,
+};
+
+/** @brief Measurement structure. */
+struct dm_result {
+	/* Status of the precedure. \a true if measurement was done correctly, \a false otherwise */
+	bool status;
+
+	/* Quality indicator */
+	enum dm_quality quality;
+
+	/* Bluetooth LE Device Address */
+	bt_addr_le_t bt_addr;
+
+	/* Mode used for ranging */
+	enum dm_ranging_mode ranging_mode;
+	union {
+		struct {
+			/* MCPD: Distance estimate based on IFFT of spectrum */
+			float ifft;
+
+			/* MCPD: Distance estimate based on average phase slope estimation */
+			float phase_slope;
+
+			/* RSSI: Distance estimate based on Friis path loss formula */
+			float rssi_openspace;
+
+			/* Best effort distance estimate */
+			float best;
+		} mcpd;
+		struct {
```

Comment:
this will result in an error once the documentation for this part of the code is being built. Doxygen/breathe/sphinx do not handle anonymous structs well and will flatten the structure in its internal representation. This leads to the "rtt" member of dist_estimates to be doubly defined (once by struct {] rtt and once by float rtt). This will result in a build failure. To resolve this, give the anonymous struct a name.
```suggestion
		struct _rtt {
```

Reply:
ACK